python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1278,30 +1264,27 @@ def get_event_tags_for_asset(
                 > datetime.utcfromtimestamp(asset_details.last_wipe_timestamp)
             )
 
-        if not perform_tag_join:
+        if not filter_tags:
             with self.index_connection() as conn:
                 results = conn.execute(tags_query).fetchall()
         else:
-            tags_query = tags_query.where(
-                db.and_(
-                    AssetEventTagsTable.c.key == tag_key, AssetEventTagsTable.c.value == tag_value
+            intersections = [
+                tags_query.where(
```

Comment:
i think this might cause some future confusion, in that sometimes, `tags_query` is the top-level query, and sometimes it's the subquery.

I'd prefer it if we just had the top-level query, and we applied the intersection and had the subquery not reuse any existing var.

So, something like this:
```
tags_query = db.select(key, value, event_id)
if filter_tags:
    intersections = [
        db.select(event_id).where(db.and_(asset_key == asset_key.to_string(), key == key, value = value))
        for key, value in filter_tags.items()
    ]
    tags_query = tags_query.where(event_id in db.intersect(*intersections))
```